CHANGELOG.md

```diff
@@ -16,6 +16,25 @@ Versioning](https://semver.org/spec/v2.0.0.html).
   is less than one second.
 - `init_expiration_time` and `update_jwt_expires_in` take `u32` instead of `i64`
   for the expiration time argument.
+- `Node` struct now has `setting` and `setting_draft` of type `NodeSetting`, and
+  `name` and `name_draft`. Upon initial insertion of `Node`, `name` must be
+  provided, as it is used as the key of `Node` in the database. `name_draft` and
+  `setting_draft` are introduced to support 2-step node-setting process, which
+  is save & apply. `name_draft` and `setting_draft` fields mean that the data
+  are only saved to the database. Once those are applied, the draft values are
+  moved to `name`, and `setting`.
+  - Renamed `updateNode` GraphQL API to `updateNodeDraft`, and modified
+    parameter types. `old` to `NodeInput`, and `new` to `NodeDraftInput`,
+    respectively with and without `setting` field.
+  - Introduced `applyNode` GraphQL API, that applies draft values to modules and
+    updates values in database. This API handles partial success of setting
+    application, which may happen when a node carries multiple modules.
+    `ApplyResult::success_modules` lists the succeeded modules' names.
+  - `graphql::event::convert_sensors` uses `Node`'s `setting` value, to retrieve
+    the hostnames of the sensors. This function is called by GraphQL APIs of
+    `EventQuery` and `EventGroupQuery`.
+  - `node_status_list` API uses `hostname` from `Node`'s `setting` field.
```

Comment:
When mentioning a GraphQL API, we should use its GraphQL API name instead of the corresponding method name. If you intend to mention the method, it should be "API method".

Reply:
done